raco/language/myrialang.py

```diff
@@ -796,12 +803,102 @@ def compileme(self):
         }
 
 
+class MyriaSamplingDistribution(algebra.UnaryOperator, MyriaOperator):
+    """A Myria SamplingDistribution operator"""
+    def __init__(self, input, sample_size, with_replacement):
+        algebra.UnaryOperator.__init__(self, input)
+        self.sample_size = sample_size
+        self.with_replacement = with_replacement
+        self.sample_type = "WR" if self.with_replacement else "WoR"
+
+    def __repr__(self):
+        return "{op}{type}{size}({inp})".format(op=self.opname(),
+                                                inp=self.input,
+                                                type=self.sample_type,
+                                                size=self.sample_size)
+
+    def shortStr(self):
+        return "{op}{type}({size})".format(op=self.opname(),
+                                           type=self.sample_type,
+                                           size=self.sample_size)
+
+    def num_tuples(self):
+        return self.input.num_tuples()
+
+    def scheme(self):
+        return self.input.scheme() + scheme.Scheme([('SampleSize',
+                                                     types.LONG_TYPE)])
+
+    def compileme(self, inputid):
+        return {
+            "opType": "SamplingDistribution",
+            "argChild": inputid,
+            "sampleSize": self.sample_size,
+            "isWithReplacement": self.with_replacement
+        }
+
+
+class MyriaSample(algebra.BinaryOperator, MyriaOperator):
+    """A Myria Sample operator"""
+    def __init__(self, left, right, sample_size, with_replacement):
+        algebra.BinaryOperator.__init__(self, left, right)
+        # with_replacement and sample_size are just used for displaying.
+        self.with_replacement = with_replacement
+        self.sample_size = sample_size
+
+    def shortStr(self):
+        samp_type = "WR" if self.with_replacement else "WoR"
```

Comment:
Make this consistent with the other `sample_type` code. Probably put the `"WR"` and `"WoR"` in exactly one place in the code